src/pages/storage/forms/UploadVolumeBackupFileForm.tsx

```diff
@@ -0,0 +1,286 @@
+  const { data: clusterMembers = [] } = useClusterMembers();
+
+  const handleSuccess = (
+    volumeName: string,
+    location: string,
+    pool: string,
+  ) => {
+    const volumeDetailURL = linkForVolumeDetail({
+      location: location,
+      name: volumeName,
+      pool: pool,
+      project: project?.name,
+      type: "custom",
+    } as LxdStorageVolume);
+
+    const message = (
+      <>
+        Created volume{" "}
+        <ResourceLink type="volume" value={volumeName} to={volumeDetailURL} />.
+      </>
+    );
+
+    const actions = [
+      {
+        label: "Configure",
+        onClick: async () => navigate(`${volumeDetailURL}/configuration`),
+      },
+    ];
+
+    toastNotify.success(message, actions);
+  };
+
+  const handleFailure = (msg: string) => {
+    toastNotify.failure("Volume creation failed.", new Error(msg));
+  };
+
+  const handleFinish = () => {
+    queryClient.invalidateQueries({
+      predicate: (query) => {
+        return query.queryKey[0] === queryKeys.volumes;
+      },
+    });
+  };
+
+  const handleUpload = (values: UploadVolumeBackupFileFormValues) => {
+    const uploadController = new AbortController();
+    setUploadAbort(uploadController);
+
+    const targetClusterMember = isCMSDriver ? values.clusterMember : "";
+
+    uploadVolume(
+      values.volumeFile,
+      values.name,
+      project?.name,
+      values.pool,
+      setUploadState,
+      uploadController,
+      targetClusterMember,
+    )
+      .then((operation) => {
+        toastNotify.info(
+          <>
+            Upload completed. Now creating volume{" "}
+            <ResourceLabel bold type="volume" value={values.name} />.
+          </>,
+        );
+
+        eventQueue.set(
+          operation.metadata.id,
+          () => {
+            handleSuccess(values.name, values.clusterMember, values.pool);
+          },
+          handleFailure,
+          handleFinish,
+        );
+
+        handleCloseModal();
+        navigate(`/ui/project/${project?.name}/storage/volumes`);
+      })
+      .catch((e: AxiosError<LxdSyncResponse<null>>) => {
+        const error = new Error(e.response?.data.error);
+        notify.failure("Volume upload failed", error);
+        formik.setSubmitting(false);
+        setUploadState(null);
+      });
+  };
+
+  const formik = useFormik<UploadVolumeBackupFileFormValues>({
+    initialValues: {
+      name: defaultVolumeName || "",
+      pool: pools?.[0]?.name,
+      volumeFile: null,
+      clusterMember: clusterMembers?.[0]?.server_name ?? "",
+    },
+    validateOnMount: true,
+    validationSchema: Yup.object().shape({
+      name: Yup.string()
+        .test(
+          ...testDuplicateStorageVolumeName(
+            project?.name || "",
+            "custom",
+            volumeNameAbort,
+          ),
+        )
+        .optional(),
+    }),
+    onSubmit: handleUpload,
+  });
+
+  const pool = pools.find((pool) => pool.name == formik.values.pool);
```

Comment:
```suggestion
  const pool = pools.find((pool) => pool.name === formik.values.pool);
```
Better use the strict type comparison.